### PR TITLE
feat: NV24 upgrade

### PR DESCRIPTION
**Pull request by LexLuthr**

### CODE REVIEW

File: api/api_chain.go [hunk 1]
```diff
@@ -67,7 +67,7 @@ type CurioChainRPC interface {
 	StateSectorPreCommitInfo(context.Context, address.Address, abi.SectorNumber, types.TipSetKey) (*miner.SectorPreCommitOnChainInfo, error)
 	StateSectorGetInfo(ctx context.Context, maddr address.Address, sectorNumber abi.SectorNumber, tsk types.TipSetKey) (*miner.SectorOnChainInfo, error)
 	StateMinerPreCommitDepositForPower(context.Context, address.Address, miner.SectorPreCommitInfo, types.TipSetKey) (big.Int, error)
-	StateMinerInitialPledgeCollateral(context.Context, address.Address, miner.SectorPreCommitInfo, types.TipSetKey) (big.Int, error)
+	StateMinerInitialPledgeForSector(ctx context.Context, sectorDuration abi.ChainEpoch, sectorSize abi.SectorSize, verifiedSize uint64, tsk types.TipSetKey) (types.BigInt, error)
```
Comment (magik6k) [inline api/api_chain.go:70]: I presume this checks the network version internally based on the tipset?

File: tasks/seal/task_submit_commit.go [hunk 2]
```diff
@@ -248,7 +251,12 @@ func (s *SubmitCommitTask) Do(taskID harmonytask.TaskID, stillOwned func() bool)
 		return false, xerrors.Errorf("could not serialize commit params: %w", err)
 	}
 
-	collateral, err := s.api.StateMinerInitialPledgeCollateral(ctx, maddr, pci.Info, ts.Key())
```
Comment (magik6k) [inline tasks/seal/task_submit_commit.go:251]: Waait how did that work with filplus deals? It couldn't be correct, right?

**Review by magik6k — APPROVED** [review]
Looks good so far, also confirmed that supraseal build still works with the new ffi

File: tasks/seal/task_submit_commit.go (hunk 2 above)
Comment (LexLuthr) [reply tasks/seal/task_submit_commit.go:251]: I don't think it did. We were sending incorrect collateral AFAIK.

File: api/api_chain.go (hunk 1 above)
Comment (LexLuthr) [reply api/api_chain.go:70]: I don't see a NV check. Why do we need it?

**Pull request merged**